packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -350,6 +350,8 @@ export function useAsyncData<
   if (import.meta.client) {
     // Setup hook callbacks once per instance
     const instance = getCurrentInstance()
+    // @ts-expect-error - internal vue property. This force vue to mark the component as async boundary client-side to avoid useId hydration issue since we treeshake onServerPrefetch
+    instance.sp = instance.sp || []
```

Comment:
```suggestion
    if (instance && fetchOnServer && options.immediate) {
      // @ts-expect-error - internal vue property. This force vue to mark the component as async boundary client-side to avoid useId hydration issue since we treeshake onServerPrefetch
      instance.sp = instance.sp || []
    }
```